### PR TITLE
Features/gatk base

### PR DESCRIPTION
**Pull request by northwestwitch**
Fixed Gatk base to inlcude only base arguments. Fixed conflics in mip.pl.
Tested on Rasta.

### CODE REVIEW

File: lib/MIP/Program/Base/Gatk.pm [hunk 1]
```diff
@@ -0,0 +1,170 @@
...
+}
+
+## Constants
+Readonly my $SPACE => q{ };
+
+sub gatk_base {
+
+## pgatk_base
+
+## Function : Perl wrapper for Gatk base. Based on Gatk v?????
```
Comment (henrikstranneheim) [inline lib/MIP/Program/Base/Gatk.pm]: 3.7

File: lib/MIP/Program/Variantcalling/Gatk.pm [hunk 2]
```diff
@@ -0,0 +1,1949 @@
...
+            defined     => 1,
+            strict_type => 1,
+            store       => \$referencefile_path
+        },
+        stderrfile_path => { strict_type => 1, store => \$stderrfile_path },
+        FILEHANDLE      => { store       => \$FILEHANDLE },
+        pedigree        => { strict_type => 1, store => \$pedigree },
+        dbsnp_file_path => { strict_type => 1, store => \$dbsnp_file_path },
+        downsample_to_coverage => {
+            allow       => qr/^\d+$/,
```
Comment (henrikstranneheim) [inline lib/MIP/Program/Variantcalling/Gatk.pm]: qr/^\d+$/ -> qr/ ^\d+$ /sxm

File: lib/MIP/Program/Variantcalling/Gatk.pm [hunk 3]
```diff
@@ -0,0 +1,1949 @@
...
+            defined     => 1,
+            strict_type => 1,
+            store       => \$referencefile_path
+        },
+        gatk_path       => { strict_type => 1, store => \$gatk_path },
+        stderrfile_path => { strict_type => 1, store => \$stderrfile_path },
+        FILEHANDLE => { store       => \$FILEHANDLE },
+        pedigree   => { strict_type => 1, store => \$pedigree },
+        downsample_to_coverage => {
+            allow       => qr/^\d+$/,
```
Comment (henrikstranneheim) [inline lib/MIP/Program/Variantcalling/Gatk.pm]: Same here

File: lib/MIP/Program/Variantcalling/Gatk.pm [hunk 4]
```diff
@@ -0,0 +1,1949 @@
...
+        },
+        stderrfile_path => { strict_type => 1, store => \$stderrfile_path },
+        FILEHANDLE        => { store       => \$FILEHANDLE },
+        pedigree          => { strict_type => 1, store => \$pedigree },
+        rscript_file_path => { strict_type => 1, store => \$rscript_file_path },
+        recal_file_path   => { strict_type => 1, store => \$recal_file_path },
+        tranches_file_path =>
+          { strict_type => 1, store => \$tranches_file_path },
+        num_cpu_threads_per_data_thread => {
+            allow       => qr/^\d+$/,
```
Comment (henrikstranneheim) [inline lib/MIP/Program/Variantcalling/Gatk.pm]: Fix throughout

File: lib/MIP/Program/Variantcalling/Gatk.pm [hunk 5]
```diff
@@ -0,0 +1,1949 @@
...
+##          : $pedigree                              => Pedigree files for samples
+##          : $recal_file_path                       => The output recal file used by ApplyRecalibration
+##          : $tranches_file_path                    => The output tranches file used by ApplyRecalibration
+##          : $num_cpu_threads_per_data_thread       => Number of CPU threads to allocate per data thread
+##          : $downsample_to_coverage                => Target coverage threshold for downsampling to coverage
+##          : $gatk_disable_auto_index_and_file_lock => Disable both auto-generation of index files and index file locking
+##          : $disable_indel_qual                    => Disable printing of base insertion and deletion tags (with -BQSR)
+##          : $logging_level                         => Set the minimum level of logging
+##          : $pedigree_validation_type              => Validation strictness for pedigree
+##          : $ts_filter_level                       => Rscript file path
```
Comment (henrikstranneheim) [inline lib/MIP/Program/Variantcalling/Gatk.pm]: Modify explanation

File: lib/MIP/Program/Variantcalling/Gatk.pm [hunk 6]
```diff
@@ -0,0 +1,1949 @@
...
+
+        push @commands,
+          q{--read_filter} . $SPACE . join $SPACE . q{--read_filter} . $SPACE,
+          @{$read_filters_ref};
+    }
+
+    if ($base_quality_score_recalibration_file) {
+
+        push @commands,
+          q{--BQSR} . $SPACE . $base_quality_score_recalibration_file;
```
Comment (henrikstranneheim) [inline lib/MIP/Program/Variantcalling/Gatk.pm]: Is this not a common parameter that should go into base?

File: lib/MIP/Program/Variantcalling/Gatk.pm [hunk 7]
```diff
@@ -0,0 +1,1949 @@
...
```
Comment (henrikstranneheim) [inline lib/MIP/Program/Variantcalling/Gatk.pm]: Is this not a common parameter that should go into base?

File: lib/MIP/Program/Variantcalling/Gatk.pm [hunk 8]
```diff
@@ -0,0 +1,1949 @@
...
+    }
+
+    if ($disable_indel_qual) {
```
Comment (henrikstranneheim) [inline lib/MIP/Program/Variantcalling/Gatk.pm]: Is this not a common parameter that should go into base?

File: lib/MIP/Program/Variantcalling/Gatk.pm [hunk 9]
```diff
@@ -0,0 +1,1949 @@
...
+
+        push @commands, q{--disable_indel_quals};
+    }
+
+    if ( @{$static_quantized_quals_ref} ) {
```
Comment (henrikstranneheim) [inline lib/MIP/Program/Variantcalling/Gatk.pm]: Is this not a common parameter that should go into base?

File: lib/MIP/Program/Variantcalling/Gatk.pm [hunk 10]
```diff
@@ -0,0 +1,1949 @@
...
+          @{$read_filters_ref};
+    }
+
+    if ($base_quality_score_recalibration_file) {
+
+        push @commands,
+          q{--BQSR} . $SPACE . $base_quality_score_recalibration_file;
+    }
+
+    if ($disable_indel_qual) {
```
Comment (henrikstranneheim) [inline lib/MIP/Program/Variantcalling/Gatk.pm]: Is this not a common parameter that should go into base?

File: lib/MIP/Program/Variantcalling/Gatk.pm [hunk 11]
```diff
@@ -0,0 +1,1949 @@
...
+
+        push @commands, q{--disable_indel_quals};
+    }
+
+    if ( @{$static_quantized_quals_ref} ) {
```
Comment (henrikstranneheim) [inline lib/MIP/Program/Variantcalling/Gatk.pm]: Is this not a common parameter that should go into base?

**Comment by northwestwitch** [comment]
Nice! Sure, and next thing tests!


Chiara

On Mon, Oct 9, 2017 at 8:42 AM, Henrik Stranneheim <notifications@github.com

File: lib/MIP/Program/Variantcalling/Gatk.pm (hunk 8 above)
Comment (northwestwitch) [reply lib/MIP/Program/Variantcalling/Gatk.pm]: It's just called by one subroutine, that's why I didn't put it among the common parameters

File: lib/MIP/Program/Variantcalling/Gatk.pm (hunk 7 above)
Comment (northwestwitch) [reply lib/MIP/Program/Variantcalling/Gatk.pm]: Depends on how you define common. In our pipeline it is only called by variantrecalibrator and the varianteval. Do you want it to be a common one?

File: lib/MIP/Program/Variantcalling/Gatk.pm (hunk 8 above)
Comment (henrikstranneheim) [reply lib/MIP/Program/Variantcalling/Gatk.pm]: I know but the deciding point is if it is part of the GATK Engine Parameters (available to all tools).

File: lib/MIP/Program/Variantcalling/Gatk.pm (hunk 7 above)
Comment (henrikstranneheim) [reply lib/MIP/Program/Variantcalling/Gatk.pm]: yes

**Comment by northwestwitch** [comment]
I see, I was thinking more in terms how common is it in our code. I'll fix
it and the rest according to GATK instead!

Chiara

On Mon, Oct 9, 2017 at 9:22 AM, Henrik Stranneheim <notifications@github.com

**Comment by henrikstranneheim** [comment]
👍

**Pull request merged**